### PR TITLE
diff: judge whether column exist when sort data

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
will panic if data don't have column

### What is changed and how it works?
add a judge, will fatal if don't have column key

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/diff/diff.go [hunk 1]
```diff
@@ -452,17 +452,32 @@ func (t *TableDiff) compareRows(ctx context.Context, chunk *ChunkRange) (bool, e
...
+
 	for i, sourceTable := range t.SourceTables {
 		rows, _, err := getChunkRows(ctx, sourceTable.Conn, sourceTable.Schema, sourceTable.Table, sourceTable.info, chunk.Where, args, ignoreCloumns, t.Collation)
 		if err != nil {
 			return false, errors.Trace(err)
 		}
-		sourceRows[fmt.Sprintf("source-%d", i)] = rows
-	}
 
-	targetRows, orderKeyCols, err := getChunkRows(ctx, t.TargetTable.Conn, t.TargetTable.Schema, t.TargetTable.Table, t.TargetTable.info, chunk.Where, args, ignoreCloumns, t.Collation)
```
Comment (IANTHEREAL) [inline pkg/diff/diff.go:463]: query many times？

**Comment by IANTHEREAL** [comment]
LGTM

**Comment by IANTHEREAL** [comment]
@csuzhangxc PTAL

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**